### PR TITLE
Added utility for creating a route based on manually adding points.

### PR DESCRIPTION
**Pull request by cafesilencio**
##Description

I often find I want a specific route in order to test various edge cases like u-turns or driving down the same street in both directions. To help create very specific routes I created a utility that will draw points and a line on the map based on long presses and using those points will fetch a map matched route that can be used for testing.

- [ ] I have added any issue links
- [ ] I have added all related labels (`bug`, `feature`, `new API(s)`, `SEMVER`, etc.)
- [ ] I have added the appropriate milestone and project boards

## Screenshots or Gifs
Screenshot_20200827-154133

## Testing

- [ ] I have tested locally (including `SNAPSHOT` upstream dependencies if needed) through testapp/demo app and run all activities to avoid regressions
- [ ] I have tested via a test drive, or a simulation/mock location app
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes

## Checklist

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have updated the `CHANGELOG` including this PR
- [ ] We might need to update / push `api/current.txt` files after running `$> make core-update-api` (Core) / `$> make ui-update-api` (UI) if there are changes / errors we're 🆗 with (e.g. `AddedMethod` changes are marked as errors but don't break SemVer) 🚀 If there are SemVer breaking changes add the `SEMVER` label. See [Metalava](https://github.com/mapbox/mapbox-navigation-android/blob/master/docs/metalava.md) docs

### CODE REVIEW

File: examples/src/main/java/com/mapbox/navigation/examples/utils/RouteDrawingUtil.kt [hunk 1]
```diff
@@ -0,0 +1,216 @@
...
+import timber.log.Timber
+
+/**
+ * A utility for drawing a line on a map and using map matching to get a route.
+ * When enable() is called the utility will listen for long click events on the map and for each point
+ * a dot will be placed on the map with a line between representing the collective points received.
+ *
+ * Calling fetchRoute() will use attempt to fetch a map matched route matching the points that
+ * have been configured.  When a route is received draw it on the map and be sure to set the route
+ * on your MapboxNavivation instance.
```
Comment (langsmith) [inline examples/src/main/java/com/mapbox/navigation/examples/utils/RouteDrawingUtil.kt:39]:
```suggestion
 * on your MapboxNavigation instance.
```

File: examples/src/main/java/com/mapbox/navigation/examples/utils/RouteDrawingUtil.kt [hunk 2]
```diff
@@ -0,0 +1,216 @@
...
+ */
+
+class RouteDrawingUtil(private val mapView: MapView) {
+
+    private val touchPoints = mutableListOf<Point>()
+
+    companion object {
+        const val LINE_LAYER_SOURCE_ID = "DRAW_UTIL_LINE_LAYER_SOURCE_ID"
+        const val LINE_LAYER_ID = "DRAW_UTIL_LINE_LAYER_ID"
+        const val LINE_END_LAYER_ID = "LINE_END_LAYER_ID"
```
Comment (langsmith) [inline examples/src/main/java/com/mapbox/navigation/examples/utils/RouteDrawingUtil.kt:51]:
```suggestion
        const val LINE_END_LAYER_ID = "DRAW_UTIL_LINE_END_LAYER_ID"
```

File: examples/src/main/java/com/mapbox/navigation/examples/utils/RouteDrawingUtil.kt [hunk 3]
```diff
@@ -0,0 +1,216 @@
...
+        const val LINE_END_SOURCE_ID = "LINE_END_SOURCE_ID"
```
Comment (langsmith) [inline examples/src/main/java/com/mapbox/navigation/examples/utils/RouteDrawingUtil.kt:52]:
```suggestion
        const val LINE_END_SOURCE_ID = "DRAW_UTIL_LINE_END_SOURCE_ID"
```

**Comment by langsmith** [comment]
I'm not sure if you're done with this pr, but I couldn't get the map matching to work. I got the black circles and yellow lines to appear just fine. Seemed that the call wasn't even being made 😕 Is it working for you @cafesilencio ? Maybe there was something weird for me with `Utils.getMapboxAccessToken(mapView.context)`.

Maybe you could add an example activity to show how to use this util class? Or add more explanation to the kdoc explanation area at the top of `RouteDrawingUtil`. Folks might have some confusion about where to init and use this class. In `onCreate()`? In `onStyleLoaded()`?  `When a route is received draw it on the map and be sure to set the route on your MapboxNavivation instance.` might not be obvious to folks.

**Comment by cafesilencio** [comment]
I hesitate to add yet another example activity for us to maintain but clearly the instructions need to be improved.  I obviously took for granted that the utility's usage would be evident.  I"ll add some more (better) instructions.
